packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -221,6 +232,26 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     });
 
     this._standard_wallets = aptosStandardWallets;
+    // Append AIP-62 compatible wallets that are not detected on the user machine
+    this.appendNotDetectedStandardSupportedWallets(this._standard_wallets);
+  }
+
+  /**
+   * A function that excludes a wallet the dapp doesnt want to include
+   *
+   * @param walletName
+   * @returns
+   */
+  private excludeWallet(walletName: string): boolean {
+    // If _optInWallets is not empty, and does not include the provided wallet,
+    // return true to exclude the wallet, otherwise return false
+    if (
+      this._optInWallets.length > 0 &&
+      !this._optInWallets.includes(walletName as AvailableWallets)
+    ) {
+      return true;
+    }
+    return false;
```

Comment:
nit: not sure if this needs to be a private method.
It's only used once and it's just simple filtering that most developer understand intuitively.

```typescript
// Filter out wallets when the user has explicitly provided an opt-in list
if (this._optInWallets.length > 0 && !this_optInWallets.includes(walletName) {
  return;
}
```